src/trigger-circleci-workflows.ts

```diff
@@ -61,17 +73,58 @@ async function triggerCircleCI(
         );
       }
     });
-  context.log.info(
-    {data},
-    `Build triggered successfully for ${context.payload['pull_request']['html_url']}`
-  );
+  context.log.info({data}, `Build triggered successfully for ${repoKey}`);
 }
 
 export function circlePipelineEndpoint(repoKey: string): string {
   return `/api/v2/project/github/${repoKey}/pipeline`;
 }
 
-async function runBot(context: probot.Context): Promise<void> {
+export function genParametersForPR(
+  config: object,
+  context: probot.Context,
+  appliedLabels: string[]
+): object {
+  context.log.debug({config, appliedLabels}, 'genParametersForPR');
+  const parameters = {};
+  const labelsToParams = config['labels_to_circle_params'];
+  for (const label of Object.keys(labelsToParams)) {
+    // ci/all is a special label that will set all to true
+    if (appliedLabels.includes('ci/all') || appliedLabels.includes(label)) {
+      parameters[labelsToParams[label].parameter] = true;
+    }
+  }
+  return parameters;
+}
+
+function genParametersForPush(config: object, context: probot.Context): object {
+  const parameters = {};
+  const labelsToParams = config['labels_to_circle_params'];
```

Comment:
The use of `labelsToParams` in this function is very obviously abusive of the original intent of this config struct, right? :)

Reply:
This is actually for more ease of use, since we re-use this variable quite a bit I thought it more appropriate to just create a variable so we didn't have to type `config['labels_to_circle_params']` out a bunch of times.